Braintree/Payments/@Public/BTPaymentProvider.h

```diff
@@ -18,6 +18,29 @@ typedef NS_ENUM(NSInteger, BTPaymentProviderType) {
     BTPaymentProviderTypeApplePay
 };
 
+/// Status of the last (or ongoing) payment method creation
+typedef NS_ENUM(NSInteger, BTPaymentProviderStatus) {
+    /// The payment method was not created yet (default status)
+    BTPaymentProviderStatusUninitialized = 0,
+
+    /// Payment method created (either PayPal, Venmo or ApplePay)
+    BTPaymentProviderStatusInitialized,
+
+    /// The payment method creator, having obtained user payment details and/or user
+    /// authorization, is now processing the results
+    BTPaymentProviderStatusProcessing,
+
+    /// Payment method failed with error
+    BTPaymentProviderStatusError,
+
+    /// Payment method creation is complete with success
+    BTPaymentProviderStatusSuccess,
```

Comment:
Maybe to be consistent with [this other enum](https://github.com/braintree/braintree_ios/blob/a1522fecf8bd14f50919594cb7b8d20e8a827d9f/Braintree/Venmo/BTVenmoAppSwitchReturnURL.h#L9), this should be `BTPaymentProviderStatusSucceeded` and `BTPaymentProviderStatusFailed`?. (also rename every *Status* to *State*?):

BTPaymentProviderStateUninitialized
BTPaymentProviderStateInitialized
BTPaymentProviderStateFailed
BTPaymentProviderStateSucceeded
BTPaymentProviderStateCanceled

.cc @burnto